riscv/src/lib.rs

```diff
@@ -195,46 +200,133 @@ runtime = {{ path = "./runtime" }}
     )
     .unwrap();
 
-    compile_rust_crate_to_riscv_asm(cargo_file.to_str().unwrap())
+    compile_rust_crate_to_riscv_asm(cargo_file.to_str().unwrap(), output_dir)
 }
 
-pub fn compile_rust_crate_to_riscv_asm(input_dir: &str) -> BTreeMap<String, String> {
-    let temp_dir = Temp::new_dir().unwrap();
+macro_rules! as_ref [
```

Comment:
Wouldn't it be better to use `.map` or something instead of introducing a custom macro?

Reply:
The problem is that the elements may have different types, so I can't write an array literal out of them to apply the map. That is why I used a macro.

All this to avoid using `.to_str().unwrap()` on a `&Path`.